### ISSUE TITLE
x/tools/imports Nested package with directory name that differs from package name is not found when fixing imports 

### ISSUE THREAD

**Issue by ZacxDev** [golang/go#46405]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go version go1.16.4 linux/amd64
</pre>

### Does this issue reproduce with the latest release?

yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/developer/.cache/go-build"
GOENV="/home/developer/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/developer/go/pkg/mod"
GONOPROXY="gitlab.innovationup.stream"
GONOSUMDB="gitlab.innovationup.stream"
GOOS="linux"
GOPATH="/home/developer/go"
GOPRIVATE="gitlab.innovationup.stream"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.16.4"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/developer/workspace/test/nested/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build1014239028=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?

Running `goimports .` against this program **will not** correct the imports

https://play.golang.org/p/_uEfcBCSS06

Running `goimports .` against this program **will** correct the imports

https://play.golang.org/p/oPWzuxbsLIK

Further, if there is only one level of nesting (helpers file is in helpers/nest-a/helpers.go), the package is found by goimports. It is only once we add another level of nesting (helpers/nest-a/nest-b/helpers.go) that the issue arises

### What did you expect to see?

I expected the imports to be correct regardless of their directory/level of nesting

### What did you see instead?

goimports does not find the package

**Comment by seankhliao** [comment]
Duplicate of #16600
